### PR TITLE
AMPI: Implement proper context activation for migration callbacks

### LINKED ISSUE #1214: AMPI_Just_migrated callbacks break using tlsglobals/isomalloc
*Original issue: https://charm.cs.illinois.edu/redmine/issues/1214*

---
AMPI doesn't properly take care of turning on/off isomalloc and tlsglobals before calling the user's justMigrated function.

### PR DESCRIPTION
**Pull request by evan-charmworks**
Fixes #1214

### CODE REVIEW

**Comment by evan-charmworks** [comment]
Ready for review. The macOS CI failure is a separate issue I am looking into.

File: src/libs/ck-libs/tcharm/tcharm_impl.h [hunk 1]
```diff
@@ -103,6 +103,8 @@ class TCharm: public CBase_TCharm
 	TCharmInitMsg *initMsg; //Thread initialization data
 	double timeOffset; //Value to add to CkWallTimer to get my clock
 
+  CkCallback resumeCallback;
```
Comment (stwhite91) [inline src/libs/ck-libs/tcharm/tcharm_impl.h]: Maybe add an inline comment describing the use of this variable

Comment (matthiasdiener) [inline src/libs/ck-libs/tcharm/tcharm_impl.h]: Indentation doesn't seem to match the surrounding code (also below).

Comment (evan-charmworks) [reply src/libs/ck-libs/tcharm/tcharm_impl.h]: Done

Comment (evan-charmworks) [reply src/libs/ck-libs/tcharm/tcharm_impl.h]: Fixed

**Pull request merged**